### PR TITLE
fix bug for OFA

### PR DESCRIPTION
**Pull request by ceci3**
pre PR: https://github.com/PaddlePaddle/PaddleSlim/pull/416

work in this PR:

- [x] support SuperEmbedding and SuperLayerNorm

- [x] support user-defined padding

- [x] fixed some layers choice in search

- [x] support input net config manually before forward

- [x] support output layers config after forward

- [x] fix when depth is float

- [x] fix some other bugs

work in next PR:
-  accuracy model
-  search process (evolution search)
-  latency predict (support dygraph latency compute)
-  flops predict(support dygraph flops compute)
-  add document

### CODE REVIEW

File: demo/one_shot/ofa_train.py [hunk 1]
```diff
@@ -59,6 +59,10 @@ def forward(self, inputs, label, depth=None):
 
 def test_ofa():
 
+    fluid.enable_dygraph()
```
Comment (wanghaoshuang) [inline demo/one_shot/ofa_train.py]: 不能用`paddle.enable_dygraph()`么？

File: paddleslim/nas/ofa/convert_super.py [hunk 2]
```diff
@@ -18,15 +18,15 @@
 import paddle
 import paddle.fluid as fluid
 from paddle.fluid import framework
-from paddle.fluid.dygraph.nn import Conv2D, Conv2DTranspose, Linear, BatchNorm, InstanceNorm
+from paddle.fluid.dygraph.nn import Conv2D, Conv2DTranspose, Linear, BatchNorm, InstanceNorm, LayerNorm, Embedding
```
Comment (wanghaoshuang) [inline paddleslim/nas/ofa/convert_super.py:20]: 顺手把`fluid`这个改掉吧，以后的代码里最好不出现fluid关键字

**Review by wanghaoshuang — DISMISSED** [review]

File: demo/one_shot/ofa_train.py (hunk 1 above)
Comment (ceci3) [reply demo/one_shot/ofa_train.py]: done，thanks

File: paddleslim/nas/ofa/convert_super.py (hunk 2 above)
Comment (ceci3) [reply paddleslim/nas/ofa/convert_super.py:20]: done, thanks。 因为ERNIE的代码用的paddle版本是1.8.5，ofa-ernie用到的文件暂时还没有隐藏fluid概念。之后会统一修改。

**Review by wanghaoshuang — DISMISSED** [review]

**Review by yukavio — APPROVED** [review]

**Pull request merged**